src/stream/src/executor/mod.rs

```diff
@@ -1046,6 +1046,29 @@ impl<M> MessageInner<M> {
 
 pub type Message = MessageInner<BarrierMutationType>;
 pub type DispatcherMessage = MessageInner<()>;
+#[derive(Debug, EnumAsInner, PartialEq, Clone)]
+pub enum MessageBatch {
+    Chunk(StreamChunk),
+    BarrierBatch(Vec<BarrierInner<BarrierMutationType>>),
```

Comment:
Do we have some empirical idea of the length of barrier batch? If it's small, shall we consider `SmallVec` here?